### PR TITLE
#1954 fix supplier requisition showing inactive indicators

### LINKED ISSUE #1950: PropTypes in SupplierRequisitionPage
## Describe the bug

- Many prop type warnings in `SupplierRequisitionPage` for non-program requisition

### To reproduce

1. Create a non-program requisition

### Expected behaviour

No prop type warnings

### Proposed Solution

Fix the prop type warnings

### Version and device info

- App version: 3.1.1 development
- Tablet model: N/A
- OS version: N/A

### Additional context
image

### LINKED ISSUE #1954: Not HIV Programs are showing HIV Indicators
## Describe the bug

Not completely sure about this one, but when creating a Requisition from a program that is not an `HIV program`, this is showing Indicators data anyway:

Program settings (See `HIV program` it's unticked)

Screen Shot 2020-01-17 at 11 49 02 AM

And the requisition is showing Indicators anyway:

Screen Shot 2020-01-17 at 11 49 28 AM

### To reproduce

1. Create a Program requisition that is not an `HIV Program`
2. See that show Indicators

### Expected behaviour

It shouldn't show Indicators (?)

### Version and device info

- App version: 3.2.0-rc3
- Tablet model: AVD 
- OS version: Android 9.1

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #1954. Branches off #1960.

## Change summary

Fixes inactive indicators being displayed. 

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Supplier requisitions for active HIV programs display HIV indicator values.
- [ ] Supplier requisitions for inactive HIV programs do not display HIV indicator values (disable `"Is HIV program"` -> resync -> create a new supplier requisition).

### Related areas to think about

N/A.

### CODE REVIEW

File: src/database/DataTypes/MasterList.js [hunk 1]
```diff
@@ -72,6 +72,14 @@ export class MasterList extends Realm.Object {
     this.addIndicator(programIndicator);
   }
 
+  /**
+   * Get all indicators currently active on this master list.
+   * @returns {Array.<ProgramIndicator>}
+   */
+  getActiveIndicators() {
```
Comment (joshxg) [inline src/database/DataTypes/MasterList.js]: why is this not a `getter`?

File: src/database/DataTypes/Requisition.js [hunk 2]
```diff
@@ -140,8 +140,12 @@ export class Requisition extends Realm.Object {
     return (this.otherStoreName && this.otherStoreName.name) || '';
   }
 
+  /**
+   * Get all indicators associated with this requisition.
+   * @returns {Array.<ProgramIndicator>}
+   */
   get indicators() {
-    return this.program?.indicators;
+    return this.program?.getActiveindicators();
```
Comment (joshxg) [inline src/database/DataTypes/Requisition.js]: `return this.program?.activeIndicators() || []` ? or nah

Comment (wlthomson) [reply src/database/DataTypes/Requisition.js]: I did consider that, but I think in the case of Realm lists, `null` testing is a bit more consistent than making the getter return two different types (object v. array).

File: src/database/DataTypes/MasterList.js (hunk 1 above)
Comment (wlthomson) [reply src/database/DataTypes/MasterList.js]: ef0a03bb4f1e9a9c5aff1de9b7688d6d00fb6cac.

please

**Pull request merged**